PL-132122 upgrade hardware to 24.11

ctheune opened the pull request:
@flyingcircusio/release-managers

## Release process

Impact:

Changelog:

- Port changes to support our physical infrastructure from the 21.05 release branch. (PL-132122)

### PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.

n/a

- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

n/a

## Security implications

- [ ] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

zagy committed:
Let gocept.net mail be handled by our main mail server.

FC-23766

sysvinit committed:
pkgs: add script for monitoring kernel routing state integrity

PL-132595

sysvinit committed:
platform: add sensu checks for underlay network functions

PL-132595

ctheune committed:
implement check to monitor important parameters

leona-ya committed:
devhost: cleanup unused VMs (PL-132737)

osnyx committed:
fc-luks check: establish sensu-check

Establish a regular sensu check on physical hosts that checks certain
LUKS header parameters of all encrypted devices, to discover unexpected
diversions.

- tie together the individual condition checks
- integrate checks into fc-luks tooling with discovery of encrypted
  volumes
- add sensu check
- extend checks with plausibility checks for proper dump input
- add unit and integration tests for checks with proper mocking

osnyx committed:
fc-luks check: integrate smoke test into NixOS tests

This makes sense to expose the check to potential changes in to output
format of `cryptsetup luksDump`.
Detailed unit tests are done against mock outputs only.

osnyx committed:
fc-luks check: only enable sensu check on hosts with keystick

Not all physical machines actually have encrypted volumes as of now,
e.g. KVM hosts. The check does not need to run on them.

osnyx committed:
fc-luks check: fix sudoers rule for sensu check

ctheune committed:
Merge pull request #1086 from flyingcircusio/leona/PL-132737-devhost-cleanup-unused-vms

devhost: cleanup unused VMs (PL-132737)

osnyx committed:
fc-luks: read PATH from config file to fix sensu check run

By running fc-luks commands as a sensu check, we suddenly do not inherit
the full system PATH anymore, breaking access to necessary external
tools used like `lvs`.
We can adopt the `fc-ceph.conf` approach already used by `fc-ceph`
subsystems, for now even the default PATH is sufficient.

Requires some extra mocking in unit tests; the NixOS integration test
now explicitly empties the path beforehands.

sysvinit committed:
pkgs: add test script for handling ping on tap interfaces

PL-132595

osnyx committed:
make nixd happy

osnyx committed:
fc-ceph.conf: refactor to enable it on non-ceph client hosts for fc-luks

osnyx committed:
[formatting] remove unnecessary indentation

sysvinit committed:
tests/frr: refactor into subtest with configuration modules

PL-132595

ctheune committed:
Merge pull request #1087 from flyingcircusio/PL-131325-fde-finalise

[21.05] full-disk encryption: finalisation, improvements, checks, routine tooling

sysvinit committed:
tests/frr: check that the fib/rib monitoring script works correctly

This test artificially causes the FIB and RIB to get out of sync, and
checks that the sensu script correctly detects these problems.

PL-132595

sysvinit committed:
tests: add frr tests to default test suite

PL-132595

ctheune committed:
Merge pull request #1088 from flyingcircusio/PL-132595-vxlan-monitor-control-plane

[21.05] Sensu monitoring for EVPN control plane

ctheune committed:
Merge pull request #1083 from flyingcircusio/PL-132849-tmp-uuids-during-restore

[21.05] restore-single-files: reduce chance of UUID collision

sysvinit committed:
Merge pull request #1080 from flyingcircusio/FC-23766-switch-mailserver

[21.05] Switch MX record for gocept.net

ctheune committed:
sensu: fix rabbitmq user updates

Due to performance reasons we did not always change the passwords,
only for new users.

This caused inconsistencies more often than we can tolerate.

This now fixes it properly by parallelizing the updates, limiting to
CPU_COUNT-1 and also deleting superfluous users.

Fixes PL-132945

osnyx committed:
Merge pull request #1091 from flyingcircusio/PL-132945-fix-rabbitmq-user-updates

[21.05] fix rabbitmq user updates for sensu

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
ceph/radosgw: improve health monitoring

We've seen instances where the / endpoint was responding but the
radosgw was completely stuck.

We hope to notice those better now with checks that actually hit a
real object.

Re PL-132070

osnyx committed:
Merge pull request #1093 from flyingcircusio/PL-132070-improve-rgw-monitoring

ceph/radosgw: improve health monitoring

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
router: use more generic netboot url and adapt menu item

Re PL-132254

sysvinit committed:
pkgs: add script for monitoring the state of the kernel neighbour cache

PL-132950

sysvinit committed:
router: add neighbour cache to telegraf and sensu

PL-132950

osnyx committed:
vendor fc.util.directory from agent to fc-ceph code

osnyx committed:
fc-ceph maintenance load-vm-images: pull from directory

The fc-ceph image loading job now pulls the list of images to process
from the directory API (needs directory changes). Which environments to
process is determined by a name glob that can be customised via a cli
argument.

`fc-ceph maintenance load-vm-images --name-glob 'some-env*'` allows
specifying which environment image(s) to pull as a manual one-off
action.

Querying Hydra directly is removed from the fc-ceph code, as this is now
done by the directory anyways.

PL-132516

sysvinit committed:
frr: raise restart limit thresholds for zebra

Give zebra a more generous limit for restarts after crashes. Allow up
to 20 restarts in 5 minutes, and wait 10 seconds between restart
attempts.

PL-132950

sysvinit committed:
router: add zebra liveness check to keepalived

Routers are only capable of fulfilling the role of primary when zebra
is running and working correctly. Add a check to the keepalived
configuration which transitions to FAULT state if zebra is not
running.

PL-132950

sysvinit committed:
router: remove dead configuration file

osnyx committed:
tests for sri<->sha256 hash conversion

osnyx committed:
fc-ceph maintenance load-vm-images: tests for data processing

ctheune committed:
Merge pull request #1095 from flyingcircusio/PL-132950-neighbour-cache-monitoring

[21.05] Monitoring and alerting for kernel neighbour cache state

osnyx committed:
test that image data parsing continues at errors

PL-132048

leona-ya committed:
devhost: improvements after deletion feature (PL-133005)

* Show online state and last deploy date in `fc-devhost ls -l`
* Fix deletion of VMs devhost cleanup timer
* Only run fc-manage in cleanup when necessary
* use new fc-manage CLI

ctheune committed:
Merge pull request #1101 from flyingcircusio/PL-132254-improve-firmware-updates

[21.05] router: use more generic netboot url and adapt menu item

osnyx committed:
rclone: add as default platform package

osnyx committed:
fc-luks: fix external header autodetection for multiple commands

I plainly forgot to pass the headerfile as an argument, despite it being
properly detected before.
This fixes the execution of `fc-luks keystore test-open` and `fc-luks
check` with external headers.

ctheune committed:
Merge pull request #1108 from flyingcircusio/os-rclone-platform-21.05

rclone: add as default platform package

ctheune committed:
Merge pull request #1097 from flyingcircusio/PL-132950-frr-keepalived-crash-tuning-integration

[21.05] zebra restart tuning and keepalived integration

ctheune committed:
Merge pull request #1110 from flyingcircusio/PL-132958-luks-headers-check

[21.05] fc-luks: fix external header autodetection for multiple commands

osnyx committed:
fc-ceph-load-vm-images: remove unnecessery sleeps

osnyx committed:
fc-ceph-load-vm-images: rework image cleanup

In the ideal case, only the single current snapshot of an image is kept.
In case of a race condition, we try again at the next run.

Always keep at least one snapshot as a safety measure against invalid
image list data.

suggestion how to make the code a bit easier to read

ctheune committed:
Merge pull request #1100 from flyingcircusio/PL-132516-load-images-api

[21.05] fc-ceph-load-vm-images: fetch image list from directory

ctheune committed:
fc.qemu: integrate update that removes superfluous XFS UUID regen

Fixes PL-133015

ctheune committed:
slight cleanup

osnyx committed:
Merge pull request #1116 from flyingcircusio/PL-133015-stop-updating-uuids-on-cold-boot

[21.05] stop updating uuids on cold boot

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
fc.qemu: update to reduce test flakiness

Re PL-133015

osnyx committed:
Merge pull request #1118 from flyingcircusio/PL-133015-improve-kvm-test-stability

fc.qemu: update to reduce test flakiness

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
cleanup: order lines

ctheune committed:
remove outdated swap code

ctheune committed:
simplify ext4 settings that are very likely outdated, rely on defaults

ctheune committed:
hardware: add nvme support

- include nvme drivers into initrd
- refactor the installer to be less picky about device names

ctheune committed:
Merge pull request #1119 from flyingcircusio/PL-132817-adapt-for-nvme-only

[21.05] Adapt netboot installer for nvme-only hosts

ctheune committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

ctheune committed:
various fixes for builds and cleanups

ctheune committed:
merge error

ctheune committed:
stop building the base qemu with the now superfluous patch.

leona-ya commented on nixos/lib/builders.nix:
```diff
@@ -0,0 +1,75 @@
+    }
+
+  */
+  writeShellApplication =
```
This shouldn't be needed anymore as `writeShellApplication` exists in upstream nixpkgs.

ctheune committed:
Merge pull request #1102 from flyingcircusio/PL-133005-devhost-deletion-improvements

[21.05] devhost: improvements after deletion feature (PL-133005)

ctheune committed:
fix a couple of evaluation errors

notes are in PL-132122

ctheune committed:
upgrade fc.qemu to fix PL-132809

osnyx committed:
Merge pull request #1123 from flyingcircusio/PL-132809-fix-freeze-thaw

[21.05] fc-qemu: fix freeze thaw

osnyx committed:
s3 management: add user management

- accounting now uses the command runner utility code

- s3 users are now managed (created, deleted and updated)

Re PL-133084

ctheune committed:
kvm: fix mtu detection for new network style, rely on bridge

fixes PL-133040

sysvinit committed:
Merge pull request #1126 from flyingcircusio/PL-133040-fix-kvm-ifup-mtu

[21.05] Fix guest TAP interface MTU setup on KVM hosts

ctheune committed:
dev: how to get an editable version of the agent

ctheune committed:
Merge pull request #1127 from flyingcircusio/dev-agent-editable

dev: how to get an editable version of the agent

ctheune committed:
physical: introduce EFI support for booting physical machines

- the flying circus platform now differentiates the `bios` and `efi`
  boot platforms using the `flyingcircus.boot-style` option.

- EFI uses systemd-boot on an UEFI vfat partition,
  BIOS remains with grub using ext4.

- routers serve iPXE in the BIOS or EFI variant based on the vendor
  class sent by the DHCP client.

- update ipxe to a newer version to properly support EFI in the sanboot
  hooks.

- The iPXE menu differentiates between BIOS and EFI and automatically
  picks up the right current flavor, but can be overriden by operators.

- converted the fc-install script from bash to Python to handle the
  more complex logic

- adapt to some IPMI password weirdnesses that may error out if
  the password has already been previously set

Fixes PL-133103

ctheune committed:
VM image migration: update fc.qemu and improve test performance and output quality.

(This likely might trigger broken tests elsewhere, where the
pytest-patterns are in use.)

Fixes PL-131857

osnyx committed:
Merge pull request #1128 from flyingcircusio/PL-133103-support-efi-on-hardware

Support EFI-based hardware install and boot

ctheune committed:
docs: remove intersphinx dependencies

Re PL-132115

ctheune committed:
switch to named constituents

Re PL-133119

ctheune committed:
Merge pull request #1139 from flyingcircusio/PL-132115-PL-133119-21.05-doc-cleanup-job-eval

docs: remove intersphinx dependencies

leona-ya committed:
Add leona root key (PL-133122)

ctheune committed:
Merge pull request #1142 from flyingcircusio/PL-133122-leona-root-key-21.05

[21.05] Add leona's root key

ctheune committed:
Use scriv for changelog management

Backport PL-133099

ctheune committed:
Merge pull request #1147 from flyingcircusio/PL-133099-scriv-backport-21.05

Use scriv for changelog management

osnyx committed:
Merge pull request #1131 from flyingcircusio/PL-131857-rbd-hdd-ssd-migration

VM image migration: update fc.qemu and improve test performance and o…

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

ctheune committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

ctheune committed:
backy: fix inheritance source

ctheune committed:
resolver: switch quad9 servers, improve debug logging

Re PL-133125

sysvinit committed:
Merge pull request #1150 from flyingcircusio/PL-133125-dns-reliability

[21.05] Switch upstream DNS resolvers to improve reliability

Commit:
release: use globs for constituents

PL-133119

ctheune committed:
Merge pull request #1154 from flyingcircusio/PL-133119-constituent-globbing-2105

[21.05][PL-133119] release: use globs for constituents

ctheune committed:
fc.qemu: update to pull in last minute fixes

Re PL-133149
Re PL-133145

osnyx committed:
Merge pull request #1157 from flyingcircusio/PL-133145-PL-133149-update-fc-qemu

fc.qemu: update to pull in last minute fixes

ctheune committed:
update to pull in another last minute fix

osnyx committed:
Merge pull request #1158 from flyingcircusio/PL-133145-PL-133149-update-fc-qemu

update to pull in another last minute fix

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
Missed a broken test.

osnyx committed:
Merge pull request #1159 from flyingcircusio/PL-133145-PL-133149-update-fc-qemu

Missed a broken test.

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
Collect changelog fragments

ctheune committed:
fc.qemu: update to avoid accidental downsizes

Re PL-133166

osnyx committed:
Merge pull request #1163 from flyingcircusio/PL-133166-qemu-downsize-fix

fc.qemu: update to avoid accidental downsizes

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
Collect changelog fragments

osnyx committed:
fc-luks: specify rekey parameters

When rekeying LUKS volumes, accidentally the default cipher and PBKDF
parameters had been used. Instead, explicitly specify our tunables that
we've decided on using.

PL-133174

osnyx committed:
fc-luks: specify rekey parameters

When rekeying LUKS volumes, accidentally the default cipher and PBKDF
parameters had been used. Instead, explicitly specify our tunables that
we've decided on using.

PL-133174

osnyx committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
apply nixpkgs scripted networking patch

During development, we continuously need to pull in a different nixpkgs
branch containing some networking patches. Until merge into
fc-24.11-dev, this branch will continuously be rebase onto the updated
nixpkgs for fc-24.11-dev.

osnyx committed:
poetry2nix: pull in via flake input

poetry2nix is added as another flake input for pulling it in regularly.
This locked info is then persisted to versions.json for pulling it in
from platform code via versions.nix.
This mirrors the mechanism we already use for nixpkgs and
nixos-mailserver.

osnyx committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
pytest_patterns, overlay: add package to all python package versions

This adds infrastructure to add additional python packages to all
python3XPackages attribute sets.
pytest_patterns is used as an example, as it was only built against the
default `python3Packages` before.

pytest_patterns currently does not build due to a 404 downlod error
regardless.

osnyx committed:
python3Packages.pytest_patterns: fix download URI computation

osnyx committed:
pytest_patterns: use version from extended python3Packages sets

osnyx committed:
python3Packages: make package set generation more readable

osnyx committed:
pre-commit reformats

osnyx committed:
physical: add dm_mirror as default kernel module

necessary for allowing several LVM disk migration scenarios

ctheune committed:
Merge pull request #1168 from flyingcircusio/dm-mirror-physical

[21.05] physical: add dm_mirror as default kernel module

ctheune committed:
s3users: introduce an additional safety-belt

We decided to simplify the directory API as we do not have a specific
use case to let storage servers discover S3 users from other clusters.

This is due to the caution to not accidentally end up deleting data
for a user in the local cluster just because someone wanted to delete
a user in a foreign cluster and due to simple bugs we end up seeing this
in the wrong cluster and act on it.

ctheune committed:
fc.qemu: get update that fixes high rbd list load

Fixes PL-133194

ctheune committed:
Merge pull request #1171 from flyingcircusio/PL-133194-fix-rbd-list-load

fc.qemu: get update that fixes high rbd list load

Commit:
Collect changelog fragments

ctheune committed:
tests: make sudo test evaluate

ctheune committed:
router: fix two evaluation assertions

- The chrony module now configures `rtcfile` and `rtcautotrim` for you.
  These options conflict with `rtcsync` and cause chrony to crash.
  Unless you are very sure the former isn't what you want, please
  remove `rtcsync` from `services.chrony.extraConfig`. Alternatively,
  disable this behaviour by `services.chrony.enableRTCTrimming =
  false;`

- The option definition `services.logrotate.extraConfig' in
  `/home/ctheune/fc-nixos/channels/fc/nixos/roles/router' no longer has
  any effect; please remove it. Modify
  services.logrotate.settings.header instead

ctheune committed:
tests: fix lamp evals, likely a merge mistake ...

ctheune committed:
tests: get kvm/ceph test evaluating again

ctheune committed:
ceph: convert deprecated logrotate config

ctheune committed:
tests: make servicecheck evaluate

ctheune committed:
tests: get frr tests evaluating again

ctheune committed:
tests: get login evaluating again

ctheune committed:
ipmitool: fix override eval

ctheune committed:
tests: get more tests evaluating

ctheune committed:
fix agent merge errors

ctheune committed:
ceph: fix snappy build

ctheune committed:
ceph/qemu: pull in builds from 21.05 to stick with existing versions

we'll replace those once we upgrade ceph and qemu in a separate step.

sysvinit committed:
nixos: add kea service

Vendored from upstream commit fe474ed61a3436644828a9709e2db6ed006f92aa,
with edits for compatiblity with 21.05.

PL-133205

sysvinit committed:
nixos: add kea service

Vendored from upstream commit fe474ed61a3436644828a9709e2db6ed006f92aa,
with edits for compatiblity with 21.05.

PL-133205

sysvinit committed:
agent: add fc-kea script to generate kea dhcp config from directory

PL-133205

sysvinit committed:
agent: add fc-kea script to generate kea dhcp config from directory

PL-133205

sysvinit committed:
router: add module for kea, use instead of dhcpd

PL-133205

sysvinit committed:
router: add module for kea, use instead of dhcpd

PL-133205

sysvinit committed:
nixos: use ip address instead of hostname for boot server

PL-133205

sysvinit committed:
nixos: use ip address instead of hostname for boot server

PL-133205

sysvinit committed:
router: set authoritative flag for dhcpv4 server

Match previous isc-dhcpd configuration.

PL-133205

sysvinit committed:
router: set authoritative flag for dhcpv4 server

Match previous isc-dhcpd configuration.

PL-133205

sysvinit committed:
tests/router: update for kea

PL-133205

sysvinit committed:
tests/router: update for kea

PL-133205

sysvinit committed:
router: correctly handle non-existent kea localconfig

PL-133205

sysvinit committed:
router: correctly handle non-existent kea localconfig

PL-133205

sysvinit committed:
changelog: update

PL-133205

sysvinit committed:
changelog: update

PL-133205

chrschm committed:
Rotate CS' root ssh key as the old one was over 5 years old

FC-41948

chrschm committed:
Rotate CS' root ssh key as the old one was over 5 years old

FC-41948

sysvinit committed:
pkgs: handle unspecified address as nexthop in check-rib-integrity

Some BGP implementations, such as the one in Nokia SR Linux, will send
the unspecified address as the nexthop in IPv4-via-IPv6 routes, in
which case we should use the peer address as the nexthop instead.

PL-133199

sysvinit committed:
pkgs: handle unspecified address as nexthop in check-rib-integrity

Some BGP implementations, such as the one in Nokia SR Linux, will send
the unspecified address as the nexthop in IPv4-via-IPv6 routes, in
which case we should use the peer address as the nexthop instead.

PL-133199

sysvinit committed:
router: fix radvd config to use correct interface name

PL-133201

sysvinit committed:
router: fix radvd config to use correct interface name

PL-133201

osnyx committed:
Merge pull request #1182 from flyingcircusio/PL-133201-radvd-fix-interface-names

[21.05] Fix radvd config to use the correct interface name

osnyx committed:
Merge pull request #1181 from flyingcircusio/PL-133199-rib-monitor-nokia-nexthop

[21.05] Fix RIB monitoring script to handle unspecified address as next hop

osnyx committed:
Merge pull request #1178 from flyingcircusio/cs/FC-41948-rotate-cs-root-key-21.05

[FC-41948] Rotate CS' root ssh key [21.05]

osnyx committed:
Merge pull request #1175 from flyingcircusio/PL-133205-dhcp-port-to-kea

[21.05] Replace dhcpd from isc-dhcp with Kea

osnyx committed:
Merge pull request #1166 from flyingcircusio/PL-133174-rekey-pbkdf

fc-luks: specify rekey parameters

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
Collect changelog fragments

osnyx committed:
Collect changelog fragments

osnyx committed:
use flake inputs for pinning and pulling fc-nixos-21.05 repos

The release/versions.json file is generated from flake inputs, manual
changes are jsut overriden at the next update.
By pinning the 21.05-related inputs via flake inputs, they will be
updated by update_nixpkgs as well.

osnyx committed:
temporarily switch to nixpkgs branch with network code rebase

That branch is currently open as a PR against our nixpkgs and is
regularly rebased onto nixos-24.11.

Necessary until the hardware and non-hardware branch can be merged.

PL-133185

osnyx committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
services.kea: unvendor module

it is available in NixOS 24.05 already

sysvinit committed:
platform: add stop commands to network flooding and learning units

The units for configuring flooding suppression and MAC learning on
bridge ports were sticky, and could only apply their settings and not
revert them.

This means that the flooding suppression configuration on routers
would differ depending on whether the host was booted with the role or
not.

PL-133202

sysvinit committed:
platform: add stop commands to network flooding and learning units

The units for configuring flooding suppression and MAC learning on
bridge ports were sticky, and could only apply their settings and not
revert them.

This means that the flooding suppression configuration on routers
would differ depending on whether the host was booted with the role or
not.

PL-133202

sysvinit committed:
router: add sensu check for flooding suppression configuratio

PL-133202

sysvinit committed:
router: add sensu check for flooding suppression configuratio

PL-133202

sysvinit committed:
changelog: update

sysvinit committed:
changelog: update

ctheune committed:
Merge pull request #1185 from flyingcircusio/PL-133202-router-flood-suppression

[21.05] Fix VXLAN interface configuration units to reset settings to defaults when deactivated

ctheune committed:
snapshot

ctheune committed:
fc.qemu: update to make compatible with new runner utility

osnyx committed:
tests: fix eval by utilising testlib.fcConfig

The network.firewall test still fails eval, but now in a more specific
way.

osnyx committed:
cleanup dev-container -> devhost migration

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

ctheune committed:
fc.qemu: update to make compatible with new runner utility

sysvinit committed:
nixos: unvendor frr service module

sysvinit committed:
nixos: adapt frr config for upstream module changes

Upstream has collapsed the separate frr daemon systemd services into a
single frr.service which uses watchfrr to manage the protocol
daemons. Adapt our frr configuration for this change.

sysvinit committed:
router: disable keepalived zebra integration

This is not compatible with the refactoring of the frr NixOS module in
24.11, so disable this for now and we can fix it later.

sysvinit committed:
tests: fix eval failure in network test

ctheune committed:
Merge pull request #1167 from flyingcircusio/PL-133084-s3-usermgmt

s3 management: add user management

ctheune committed:
fix mkfs.xfs compatibility with 5.15 kernels

osnyx committed:
images: fix build

image builds do not yet have enc information about all interfaces

osnyx committed:
tests.router: fix eval

sysvinit committed:
nixos: use the correct names for the frr daemon options

sysvinit committed:
tests/frr: fix tests for 24.11

osnyx committed:
release getDottedPackageNames: add debug traces to observe eval locations

Useful for determining which package set the eval has recursed into or
on which package name it is falling over.

osnyx committed:
python38*: work around eval failures

Eval was failing due to recursing into all python packages, trying to
eval them with python38. But some of them are explicitly marked to not
build with that python version, throwing an eval failure.

We don't need to recurse into the package set with our release logic,
because the packages important to us are mentioned somewhere else
explicitly and then pull in the specific python packages relevant to
them.

ctheune committed:
backy: explicitly enable and configure mdadm

ctheune committed:
backy: fix build

ctheune committed:
backy: fixes for 24.11

- xfs compatibility
- ensure unmounting /mnt/keys works even for multiple layers of mounts
- adapt to more strict md device naming

ctheune committed:
ceph and fde: refactor tests into smaller chunks

osnyx committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
tests.router: fix import of helpers

ctheune committed:
kvm_host: fix fc-qemu-scrub not working after reboots

The timer was missing an OnBootSec which went unnoticed because KVM
servers have rarely seen reboots in the last months.

PL-133211

osnyx committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
Update nixpkgs and other flake inputs

osnyx committed:
tests.router: fix system path initialisation

The `Machine.script` property is not accessible from the python test
runner anymore. It was used to retrieve the top-level system nix store
path without any specialisation being active.
The approach was already a bit cursed, as it involved regex-parsing the
test script for the kernel path.

Instead, a similarly cursed approach generates python statements from
Nix code that initialise the python instances with appropriate values
derived from the Nix config.

The tests still fail and automatic specialisation switching does not
work, but I verified that the store paths used by the python code are
the correct ones.

osnyx committed:
tests.router: fix mocked profile registration

necessary to find specialisations to switch to

sysvinit committed:
tests.router: fix kea config for compatibility with 2.6.0 and later

sysvinit committed:
agent: add explicit subnet id generation to kea config

As of Kea 2.6.0, automatic generation and assignment of subnet
identifiers is no longer supported. Our Kea config generation script
relied on the automatic assignment so we didn't have to do that
ourselves, however this point is now moot.

This uses a counter to assign an index to each IP subnet as the config
is generated. Adding or removing subnets from the directory will cause
the IDs to change and potentially invalidate the lease database files,
but given how rarely we change our IP space and that we mostly have
DHCP service in order to hand out static leases, this isn't a problem
worth worrying about.

osnyx committed:
tests.router: skip type check for all tests

necessary due to import of external python modules

osnyx committed:
tests.router.interactive: fix variable renaming

Commit:
pkgs/overlay: nix: 2.3.16 -> 2.3.18

PL-133165

Motivation here is to be able to substitute zstd-compressed paths to
increase the total throughput of our Hydra.

The entire diff from 2.3.16 to 2.3.18 can be reviewed at
https://github.com/NixOS/nix/compare/2.3.16...2.3.18

I went through the changes and didn't notice any notable changes that
I'd expect to be an issue.

osnyx committed:
tests.network: make `router` utilise testlib

The `router` node now utilises testlib for configuring its vlans. This
fixes some missing address assignments.

ctheune committed:
fix physical installer test

osnyx committed:
tests.network.routes: restructure enc parameters

This gets us a little bit further in the tests, which still fail though.
vlans are now enabled through testlib only, enc parameters merged and
inserted properly.

ctheune committed:
fix webproxy test and improve interface rename error messages

osnyx committed:
Merge pull request #1202 from flyingcircusio/PL-133165-nix-2.3-backport

[21.05][PL-133165] pkgs/overlay: nix: 2.3.16 -> 2.3.18

ctheune committed:
fix servicecheck tests

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
Merge pull request #1197 from flyingcircusio/PL-133211-fix-qemu-scrub-timer

kvm_host: fix fc-qemu-scrub not working after reboots

osnyx committed:
Merge branch 'fc-21.05-dev' into fc-21.05-staging

osnyx committed:
Collect changelog fragments

ctheune committed:
fix routing tests - clean up address collisions with upstream harness

ctheune committed:
testlib: make the net filter easier to understand

ctheune committed:
get nfs tests working

osnyx committed:
nixpkgs: bump to include python wheel fix

osnyx committed:
tests.network: correct remark on upstream behaviour

We cannot override the test harness node IDs as they are read-only.

ctheune committed:
frr: micro upgrade and make tests less flaky

ctheune committed:
mailserver: fix tests

ctheune committed:
frr/test: remove identical mac from bridge

AFAICT this isn't how we set things up in production and it causes
spurious/flaky issues in the FIB.

osnyx committed:
nixpkgs: update

osnyx committed:
Merge branch 'fc-24.11-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
Merge branch 'fc-21.05-dev' into PL-132122-upgrade-hardware-to-24.11

osnyx committed:
tests.nfs: increase shutdown timeout

We do not enforce or guarantee a certain specific reboot time. Under
load, the test slightly missed the 10 seconds timeout. During multiple
invocations I could not make out any specific structural problem with
NFS unmounts during shoutdown, so let's just be a bit more generous here
with the timeouts.

ctheune committed:
allow bootstrapping a dev environment from 21.05

ctheune committed:
physical: fix various smaller errors while testing on real hardware

ctheune committed:
ensure more packages are built on hydra

ctheune committed:
downgrade consul

ctheune committed:
qemu: get test in a better shape (not quite there yet)

ctheune committed:
add promtail to important packages

ctheune committed:
typos: chrony and kea

ctheune committed:
static: record site number for saltlabs

ctheune committed:
hardware: fix kea config generation and various warnings

ctheune committed:
hotfix: introduce new v6 networks for lab and whq

osnyx committed:
important-packages: improve percona name references

Since 24.11, the innovation and lts names are deprecated aliases.

ctheune committed:
adapt keepalived config for new networks

PL-133260

osnyx committed:
network: bugfixes for -netdev setup script
